### ISSUE TITLE
Serializing workflow AST to JSON

### ISSUE THREAD

**Issue by behnazh-w** [rhysd/actionlint#191]
I'm trying to serialize the workflow AST generated by `actionlint` to JSON using the code below:
```go
workflow, parse_err := actionlint.Parse(data)
result, json_err := json.Marshal(workflow)
```

Which works fine in most cases except for the following example for which I get `json: unsupported type: map[*actionlint.String]*actionlint.WorkflowCallEventInput`:
```yaml
name: Example
on:
  workflow_call:
    outputs:
      foo:
        description: Foo
        value: ${{ jobs.example.outputs.foo }}

jobs:
  example:
    outputs:
      foo: ${{ steps.example.outputs.foo }}
    name: Example 
    runs-on: ubuntu-latest
    steps:
    - name: example
      id: example
      shell: bash
      run: |
        echo "::set-output name=foo::bar"
```

Do you have any plans to make the workflow AST serializable? If not, could you please provide guidance on creating a PR?

**Comment by rhysd** [comment]
> Do you have any plans to make the workflow AST serializable?

No, I don't have any plan to add such feature because it is not necessary for actionlint.

**Comment by behnazh-w** [comment]
The issue is fixed in `v1.6.20`. I suspect [this conversion](https://github.com/rhysd/actionlint/commit/d8931382faeab80948b2cdf67a6654aca9ec2958#diff-0945956cd950d047e7fa866dcf34fee7d29cbce277d4e5e5cd621f2213f3be6cR281) has fixed the marshaling issue.
